node/pegnet/snapshot.go

```diff
@@ -0,0 +1,148 @@
+	if err != nil {
+		return err
+	}
+
+	_, err = tx.Exec(`DELETE FROM snapshot_current WHERE peg_balance >= 0`)
+	if err != nil {
+		return err
+	}
+
+	_, err = tx.Exec(`INSERT INTO snapshot_current SELECT * FROM pn_addresses`)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SelectSnapshotBalances returns a map of all valid PTickers and their associated
+// snapshot balances for the given address. The snapshot balance is the minimum of the past, and current
+// snapshot.
+// You must provide the table to query on
+func (Pegnet) SelectSnapshotBalances(tx QueryAble) ([]BalancesPair, error) {
+	// This query merges all addresses that exist in both snapshots. The balance
+	// in the column is the minimum balance of the 2 snapshots. If the
+	// address does not exist in either column, it will not be present in the
+	// resulting query.
+	query := fmt.Sprintf(`SELECT %s
+		FROM snapshot_past as sn_past
+       		INNER JOIN snapshot_current as sn_current
+		WHERE sn_past.address = sn_current.address;`, snapshotMinSelectCols)
+	rows, err := tx.Query(query)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, nil
+		}
+		return nil, err
+	}
+	defer rows.Close()
+
+	var res []BalancesPair
+	for rows.Next() {
+		var bp BalancesPair
+		bp.Balances = make([]uint64, int(fat2.PTickerMax))
+
+		var id int
+		var address []byte
+		err = rows.Scan(
+			&id,
+			&address,
+			&bp.Balances[fat2.PTickerPEG],
+			&bp.Balances[fat2.PTickerUSD],
+			&bp.Balances[fat2.PTickerEUR],
+			&bp.Balances[fat2.PTickerJPY],
+			&bp.Balances[fat2.PTickerGBP],
+			&bp.Balances[fat2.PTickerCAD],
+			&bp.Balances[fat2.PTickerCHF],
+			&bp.Balances[fat2.PTickerINR],
+			&bp.Balances[fat2.PTickerSGD],
+			&bp.Balances[fat2.PTickerCNY],
+			&bp.Balances[fat2.PTickerHKD],
+			&bp.Balances[fat2.PTickerKRW],
+			&bp.Balances[fat2.PTickerBRL],
+			&bp.Balances[fat2.PTickerPHP],
+			&bp.Balances[fat2.PTickerMXN],
+			&bp.Balances[fat2.PTickerXAU],
+			&bp.Balances[fat2.PTickerXAG],
+			&bp.Balances[fat2.PTickerXBT],
+			&bp.Balances[fat2.PTickerETH],
+			&bp.Balances[fat2.PTickerLTC],
+			&bp.Balances[fat2.PTickerRVN],
+			&bp.Balances[fat2.PTickerXBC],
+			&bp.Balances[fat2.PTickerFCT],
+			&bp.Balances[fat2.PTickerBNB],
+			&bp.Balances[fat2.PTickerXLM],
+			&bp.Balances[fat2.PTickerADA],
+			&bp.Balances[fat2.PTickerXMR],
+			&bp.Balances[fat2.PTickerDASH],
+			&bp.Balances[fat2.PTickerZEC],
+			&bp.Balances[fat2.PTickerDCR],
+			// V4 Additions
+			&bp.Balances[fat2.PTickerAUD],
+			&bp.Balances[fat2.PTickerNZD],
+			&bp.Balances[fat2.PTickerSEK],
+			&bp.Balances[fat2.PTickerNOK],
+			&bp.Balances[fat2.PTickerRUB],
+			&bp.Balances[fat2.PTickerZAR],
+			&bp.Balances[fat2.PTickerTRY],
+			&bp.Balances[fat2.PTickerEOS],
+			&bp.Balances[fat2.PTickerLINK],
+			&bp.Balances[fat2.PTickerATOM],
+			&bp.Balances[fat2.PTickerBAT],
+			&bp.Balances[fat2.PTickerXTZ],
+			// v5 Additions
+			&bp.Balances[fat2.PTickerHBAR],
```

Comment:
There are 20 new pAssets in staging, please sync from latest staging, @sigrlami

Reply:
@StarNeit updated currencies list with proper tickers. Thanks